gnocchi/cli.py

```diff
@@ -228,7 +240,35 @@ def run_watchers():
                       'partitioning. Retrying: %s', e)
             raise tenacity.TryAgain(e)
 
-    def _get_tasks(self):
+        filler = threading.Thread(target=self._fill_sacks_to_process)
+        filler.daemon = True
+        filler.start()
+
+    @utils.retry
+    def _fill_sacks_to_process(self):
+        n = notifier.get_driver(self.conf.notifier)
+        if not n:
+            LOG.warning("Notifier not configured. "
+                        "Metric processing will not be real-time.")
+            # Terminate the thread
+            return
+
+        try:
+            for sack in n.iter_on_sacks_to_process():
+                if sack in self._get_sacks_to_process():
+                    LOG.debug(
+                        "Got notification for sack %s, waking up processing",
+                        sack)
+                    self.sacks_with_measures_to_process.add(sack)
```

Comment:
curious, but isn't it simpler to just have this trigger it's on code to process sack rather than hijacking the periodic polling job. seems like we end up with a lot of conditionals in _run_job to discover what triggered it.  we wouldn't have to worry about ensuring sacks_with_measures_to_process is thread safe... 

also, when we run self.sacks_with_measures_to_process.discard(s) below, that doesn't seem like it's safe since we can easily just process a sack with 100metrics but  during processing a bunch of new measures for metrics processed earlier in sack our added.